### DISCUSSION TITLE
毕业论文 - CS自学指南

### DISCUSSION THREAD

**Discussion by giscus** [PKUFlyingPig/cs-self-learning#184] [Announcements]
# 毕业论文 - CS自学指南

CS自学指南

https://csdiy.wiki/%E5%BF%85%E5%AD%A6%E5%B7%A5%E5%85%B7/thesis/

**Comment by Unik-lif** [comment]
小声bb：OverLeaf也蛮香的，套用自己学校已有的模板后可以节省很多时间，导师如果要改你的文章也比较方便，唯一的弱点就是慢和不稳定。

**Reply by lrg11** [reply]
求LaTeX的论文模板，PPT模板

**Reply by digitzh** [reply]
LaTeX Templates - The Best Source of Free Quality LaTeX Templates
https://www.latextemplates.com

**Reply by hantang** [reply]
需要模板可以参考这个仓库：https://github.com/hantang/latex-templates
按学校和模版类型收录较多的学位论文模板，主要是latex，也有word格式的。

**Reply by Mike4029** [reply]
现在不慢了

**Comment by NinZeige** [comment]
这些英语资源对读者的英语能力提出了比较大的要求, 可能会让有志于深入学习但是英语能力不足的读者望而却步, 不知道有没有人愿意提供这些优秀课程的翻译, 或者是成立汉化小组的形式提供翻译, 这样可以让更多人受益

**Reply by jwj1342** [reply]
或许可以看看这个仓库：https://github.com/apachecn/home

**Comment by wengao59** [comment]
体系结构方面，我觉得南京大学计算机系统基础的PA也蛮不错的，不过就是要学的东西很多，做做还挺有收获的。推荐一下。

**Comment by Chrunge** [comment]
正好要写毕业论文了

**Comment by s1xs1x** [comment]
有类似的EE自学指南吗？

**Comment by Minamoto25** [comment]
请问大家有没有什么英语学习方法吗（小声），比如怎么解决有些句子理解不确切，以及怎么练听力可以达到跟上没有字幕的课程的程度。

**Reply by raindrop2508** [reply]
+1我也想知道

**Reply by yaofan97** [reply]
这个问题有点太宽泛了 不知道题主本身的英语水平 但要快速提升英语水平的话可以选一本已经熟悉的编程语言来阅读，比如Java可以选Introduction to Java Programming.这是针对阅读，遇到不认识的单词可以添加到Anki或者任意一款Flashcard软件以便后期回顾。尽量电子化阅读，方便整理和标记生词。对应的生词添加自动化流程可以去知乎查。听力的话可以选择自己喜欢内容的Podcast，例如WSJ Tech News Briefing,尽量选择自己感兴趣的材料,也可以去YouTube找TED视频shadowing,就是跟读。如果本身喜欢看技术类视频那更好了，同样可以去YouTube搜索前后端的视频来看，一开始可以打开英文字幕，熟悉之后关闭字幕。语言学习关键就在于决心和坚持。知乎也还有很多学习方法，仅供参考。

**Reply by Unik-lif** [reply]
其实我感觉不要在意语言，死磕文本材料或者相关材料就行，一开始很痛苦，但是坚持一个月后不看英文都会觉得有点不适应（）
anyway，pace yourself.

**Reply by Tentabrella** [reply]
个人认为技巧上, 重点是要靠猜, 也就是别人说每一句话之前要对其准备接下来叙述的内容有一定的预测.

**Reply by akaneyinyue** [reply]
有一个叫otter的app很好用，可以实时转写英语音频，初期实在听不懂又没字幕的话可以试试~

**Reply by s1xs1x** [reply]
看课本的话不是特别需要英语水平，只是你不够习惯或者那个知识本身就比较难以理解，用中文也是看不懂的。

**Reply by HiYulia** [reply]
podcast是我推荐的一个方法，或许一开始有点痛苦，因为听不懂，老需要停下来，用语音查字典。但逐渐坚持下来，有一天会再某一刻发现自己可以基本上听懂了。而且因为是语音，随时随地都可以打开。

**Comment by raindrop2508** [comment]
恭喜毕业哈

**Comment by LittleYe233** [comment]
大一成绩不够没法转系/辅修的 EEer，只能靠自学作为慰藉。

我不认为国内大学在 CS 的本科生教学上能超过国外，但有几大原因让我执意转系，这里命中了两点：

- > 交流沟通的不便
- > 而这一切，没有学分，没有绩点，没有老师，没有同学

EE 的课程难度也大，这对于自学 CS 会带来较大的影响（大佬除外；不过若我真是大佬，不如高考多考个几分）；而且非 CS 专业想直接联系相关课程老师提问也不容易；交流群信息爆炸，且容易混杂无关吹水的消息，不适合作为答疑手段（个人倾向于 forum）……

不管怎么说，现在只能靠自学度日了。

**Reply by sid1022** [reply]
题主大几啦？我情况跟你相似EE专业自学CS，但目前马上升大三了，兼顾着课业成绩和自学CS 以及可能要参加一些比赛的压力确实大

**Reply by LittleYe233** [reply]
马上升大二。兼修CS+比赛（考虑冲保研）确实挺累。我看同校同年级一个CS专业的参加编译器的比赛一天要花几个小时在比赛上，如果还有学业压力影响是蛮大的。

**Reply by Subentropist** [reply]
哎，我们都是一个级的，都跨专业失败，都要学cs，好巧😂

**Comment by Ericlalalll** [comment]
大佬，有考虑过建一个discord群吗？然后根据不同的学校分channel，再根据课程建subchannel ，这样大家可以更方便的交流当前所学课程

**Reply by LittleYe233** [reply]
个人认为考虑到国内 discord 用户基数，直接按照课程划分 channel，无需再按照学校分，否则很多 subchannel 可能半天等不到一个人。

**Reply by HiYulia** [reply]
加我一个

**Reply by morninghut** [reply]
求邀请链接

**Reply by larson202088** [reply]
我也要参加

**Reply by Yihua339** [reply]
我也要参加

**Reply by aaaM2002** [reply]
过去这么久。请问discord已经建成了吗？如果有的话可以回复一下吗，谢谢！

**Comment by Aniurm** [comment]
毕业快乐！
自学指南让我受益匪浅，感谢你

**Comment by riverback** [comment]
请问有没有什么Clustering相关的书籍或者课程推荐呢🥺并不是信科出身，但是目前科研要用到，想趁着暑假稍微系统的学习一下

**Comment by Snorlax0124** [comment]
请问自学系列课程对计算机有什么要求吗？
M1款Mac可以胜任吗？

**Reply by Chrunge** [reply]
用Intel或AMD芯片吧，听说M1有很多兼容性问题。

**Reply by suntong30** [reply]
涉及到装x86虚拟机就不太行。如果课程实验有docker的就可以（e.g. cs162）

**Reply by zxuhan7** [reply]
自己14pro没有太多兼容性问题，即便有一些环境配置上的小问题都可以Google找到答案，很多课程里教授自己的讲课电脑就是mac

**Comment by DawnMercer-able** [comment]
极棒的cs自学指南

**Comment by lrg11** [comment]
请问有LaTeX模板吗？另外想求beamer PPT模板，谢谢

**Reply by digitzh** [reply]
LaTeX Templates - The Best Source of Free Quality LaTeX Templates
https://www.latextemplates.com

**Comment by CHUCHENN** [comment]
。

**Comment by notfresh** [comment]
同学你读研了还是直博了啊

**Reply by notfresh** [reply]
说不定咱们可能碰到，反正我是很佩服你了

**Comment by bartorat** [comment]
才发现大佬写这个的时候居然还是个本科生吗，好厉害。

**Comment by Jennifer1996** [comment]
竟然是22年才毕业的本科学弟，年少有为，很感谢能有这些资料帮助来夯实基本功

**Comment by JiasenLee** [comment]
推荐一个科大维护的毕业论文LaTeX模板列表：[List of LaTeX template for theses](https://github.com/ustctug/awesome-latex-thesis)，可以在这里找找看有没有自己的学校。

**Comment by Bit0r** [comment]
都需要自己调格式，最起码Word所见即所得。对比之下Latex的编程语法还不如html，真要代码写论文不如Markdown+AsciiMath+Pandoc，或者直接用typst。

**Comment by sitxiaochen** [comment]
生产学术垃圾哈哈哈哈哈哈哈哈哈哈哈

**Comment by lil-Journey** [comment]
自建日本节点服务器 月租5元不限流量 本人也在用 失效时可联系人工修复 qq:1165116287

**Comment by lil-Journey** [comment]
自建日本节点服务器 月租5元不限流量 本人也在用 失效时可联系人工修复 qq:1165116287
